cmd/journald.go

```diff
@@ -0,0 +1,189 @@
+// Copyright © 2017 Mesosphere Inc. <http://mesosphere.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/user"
+	"strconv"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	groupReadBit = 1 << 5
+	groupExecBit = 1 << 3
+
+	// systemdJournal is a linux system group.
+	systemdJournal = "systemd-journal"
+)
+
+var (
+	// the default location for journal is /var/log/journal, however if the folder is there,
+	// journald will write to /run/log/journal in a nonpersistent way.
+	systemJournalPaths = []string{"/var/log/journal", "/run/log/journal"}
+
+	userJournalPath string
+)
+
+type (
+	grp struct {
+		id   uint32
+		name string
+	}
+	fn func(string, uint32, map[string]uint32) error
```

Comment:
It seems a little weird to use the name `fn` at the package level.  This means that it will be shared amongst all of the other cmds in the package.  Would it be better to name it something more specific to this check?